addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200ThingHandler.java

```diff
@@ -0,0 +1,451 @@
+                        "Current selected weekday for cycle selection", "Weekday", true, true, state, "");
+                subChannels.add(newChannel);
+
+                state = new StateDescription(BigDecimal.valueOf(0), null, BigDecimal.valueOf(1), "%d", true, null);
+                newChannel = createChannel(new ChannelTypeUID(thing.getUID().getAsString() + ":" + "nbrCycles"),
+                        new ChannelUID(thing.getUID(), "nbrCycles"), service + "/" + "nbrCycles", "Number",
+                        currentPathName, "Number of switching cycles", "Number", true, true, state, "");
+                subChannels.add(newChannel);
+
+                state = new StateDescription(BigDecimal.valueOf(0), null, BigDecimal.valueOf(1), "%d", false, null);
+                newChannel = createChannel(new ChannelTypeUID(thing.getUID().getAsString() + ":" + "cycle"),
+                        new ChannelUID(thing.getUID(), "cycle"), service + "/" + "cycle", "Number", currentPathName,
+                        "Current selected cycle", "Cycle", true, true, state, "");
+                subChannels.add(newChannel);
+
+                state = new StateDescription(BigDecimal.valueOf(0), null, BigDecimal.valueOf(1), "%d minutes", false,
+                        null);
+                String posName = thing.getProperties().get(SWITCH_PROGRAM_POSITIVE);
+                newChannel = createChannel(new ChannelTypeUID(thing.getUID().getAsString() + ":" + posName),
+                        new ChannelUID(thing.getUID(), posName), service + "/" + posName, "Number", currentPathName,
+                        "Positive switch of the cycle, like 'Day' 'On'", posName, true, true, state, "minutes");
+                subChannels.add(newChannel);
+
+                String negName = thing.getProperties().get(SWITCH_PROGRAM_NEGATIVE);
+                newChannel = createChannel(new ChannelTypeUID(thing.getUID().getAsString() + ":" + negName),
+                        new ChannelUID(thing.getUID(), negName), service + "/" + negName, "Number", currentPathName,
+                        "Negative switch of the cycle, like 'Night' 'Off'", negName, true, true, state, "minutes");
+                subChannels.add(newChannel);
+
+            }
+            ThingBuilder thingBuilder = editThing();
+            List<Channel> actChannels = thing.getChannels();
+            for (Channel channel : actChannels) {
+                thingBuilder.withoutChannel(channel.getUID());
+            }
+            thingBuilder.withChannels(subChannels);
+            updateThing(thingBuilder.build());
+            isInited = true;
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        logger.debug("Handle removed");
+        this.updateStatus(ThingStatus.REMOVED);
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
```

Comment:
Imho this would by the way be a better place to handle the removal of the parent bridge, if its needed.

Note that if you would remove this methods you would get the first two cases for free, I don't thing UNKNOWN is that important so I would prefer to do that otherwise, please remove the first two cases and call the super method instead.

```java
super.bridgeStatusChanged(bridgeStatusInfo);
... the check for unknown ...
```

Reply:
OK, done.